models/time_cache.cr

```diff
@@ -0,0 +1,26 @@
+struct TimeCache(K, V)
```

Comment:
Yes, with this we are making at most 144 requests per day, I think GitHub can handle that :-)

About caching just one set of results and doing the sorting, I thought about that, but right now you are only fetching the first 100 repos, and we have more than 100 already (yay!), so for now it makes sense to cache these things separately.